dissect/util/compression/lzvn.py

```diff
@@ -0,0 +1,237 @@
+            opc_len = 3
+            L = _extract(opc, 8, 6, 2)
+            M = _extract(opc, 8, 3, 3) + 3
+
+            if src_size <= opc_len + L:
+                break
+
+            (D,) = _H.unpack(src.read(2))
+
+        elif opc in OP_PRE_D:
+            # "previous distance": This opcode has the structure LLMMM110, where the
+            # length of the literal (0-3 bytes) is encoded by the high 2 bits of the
+            # first byte. We first extract the literal length so we know how long
+            # the opcode is, then check that the source can hold both this opcode and
+            # at least one byte of the next (because any valid input stream must be
+            # terminated with an eos token).
+            opc_len = 1
+            L = _extract(opc, 8, 6, 2)
+            M = _extract(opc, 8, 3, 3) + 3
+
+            if src_size <= opc_len + L:
+                break
+
+        elif opc in OP_SML_M:
+            # "small match": This opcode has no literal, and uses the previous match
+            # distance (i.e. it encodes only the match length), in a single byte as
+            # 1111MMMM.
+            opc_len = 1
+            L = None
+            M = _extract(opc, 8, 0, 4)
+
+            if src_size <= opc_len:
+                break
+
+        elif opc in OP_LRG_M:
+            # "large match": This opcode has no literal, and uses the previous match
+            # distance (i.e. it encodes only the match length). It is encoded in two
+            # bytes as 11110000 MMMMMMMM.  Because matches smaller than 16 bytes can
+            # be represented by sml_m, there is an implicit bias of 16 on the match
+            # length; the representable values are [16,271].
+            opc_len = 2
+            L = None
+
+            if src_size <= opc_len:
+                break
+
+            M = src.read(1)[0] + 16
+
+        elif opc in OP_SML_L:
+            # "small literal": This opcode has no match, and encodes only a literal
+            # of length up to 15 bytes. The format is 1110LLLL LITERAL.
+            opc_len = 1
+            L = _extract(opc, 8, 0, 4)
+            M = None
+
+        elif opc in OP_LRG_L:
+            # "large literal": This opcode has no match, and uses the previous match
+            # distance (i.e. it encodes only the match length). It is encoded in two
+            # bytes as 11100000 LLLLLLLL LITERAL.  Because literals smaller than 16
+            # bytes can be represented by sml_l, there is an implicit bias of 16 on
+            # the literal length; the representable values are [16,271].
+            opc_len = 2
+
+            if src_size <= opc_len:
+                break
+
+            L = src.read(1)[0] + 16
+            M = None
+
+        elif opc in OP_NOP:
+            opc_len = 1
+            L = None
+            M = None
+
+            if src_size <= opc_len:
+                break
+
+        elif opc in OP_EOS:
+            opc_len = 8
+
+            if src_size <= opc_len:
```

Comment:
The reference uses `<` here instead of `<=`
```suggestion
            if src_size < opc_len:
```